### PR TITLE
Nightly Build setup

### PR DESCRIPTION
**Pull request by anirudhacharya**
### Description of changes:

changes to enable nightly build.

#### Style and formatting:

I have run `pre-commit install` to ensure that auto-formatting happens with every commit.

#### Issue number, if available

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
 import os
 import sys
+from datetime import date
 
 # Third Party
 import setuptools
 
-exec(open("smdebug/_version.py").read())
+# First Party
+import smdebug
```
Comment (jarednielsen) [inline setup.py]: We can't import smdebug in setup.py - this is where we build the package in the first place. Why are we switching this around?

Comment (anirudhacharya) [reply setup.py]: i can switch it back, but this works too. i have checked.

File: setup.py [hunk 2]
```diff
@@ -29,8 +46,10 @@ def build_package(version):
     setuptools.setup(
         name="smdebug",
         version=version,
+        long_description="\n".join(DOCLINES[2:]),
+        long_description_content_type="text/x-rst",
         author="AWS DeepLearning Team",
-        description="Automated debugging for machine learning",
+        description=DOCLINES[0],
```
Comment (jarednielsen) [inline setup.py]: What are we changing this to? Someone might change the README later and unintentionally mess up the description. Might be best to just leave the description hard-coded.

File: setup.py (hunk 1 above)
Comment (jarednielsen) [reply setup.py]: Interesting. As long as you've tested it in a fresh environment where smdebug isn't already installed - just never seen it done this way before.

File: setup.py [hunk 3]
```diff
@@ -82,4 +101,15 @@ def git(*args):
 
     sys.exit(1)
 
-build_package(version=CURRENT_VERSION)
+
+def detect_smdebug_version():
+    sys.path.append(os.path.abspath("smdebug"))
```
Comment (jarednielsen) [inline setup.py]: Why is this necessary?

File: setup.py (hunk 2 above)
Comment (anirudhacharya) [reply setup.py]: this change was done on Amol's request. @leleamol

File: setup.py (hunk 3 above)
Comment (anirudhacharya) [reply setup.py]: removed.

File: setup.py [hunk 4]
```diff
@@ -1,13 +1,30 @@
+#!/usr/bin/env python
+""" Amazon SageMaker Debugger is an offering from AWS which help you automate the debugging of machine learning training jobs.
```
Comment (Vikas-kum) [inline setup.py]: s/help/helps
